tests/Fixer/CastNotation/ShortScalarCastFixerTest.php

```diff
@@ -69,7 +69,7 @@ public function provideFixCases()
         }
     }
 
-    public function provideFixDeprecatedCases()
+    public function provideFixDeprecatedCases(): \Generator
```

Comment:
```suggestion
    public function provideFixDeprecatedCases(): iterable
```

Reply:
- i see no value in saying "array" instead of "iterable". what's the benefit?
- `Generator` *are* `iterables` (ref https://3v4l.org/kZjSo )

Reply:
so I can refactor later, so I see `: array` in the test and rewrite to `\Generator`, if we make everything `iterables` I cannot do that as easy

Reply:
> so I can refactor

this is my reason too - so I can refactor and not have to change return `type`.

Reply:
There was no return type, so not sure what you didn't want to change?

Reply:
I want them all to be `iterable`.

Reply:
I don't, especially the ones where the caller uses array-access it is nice to keep array for example, and I like to refactor later without going through all the return types again and not only array

Reply:
I'm aiming to refactor most of them in an automated way - especially the providers starting with a `return` should not be hard to handle.

Reply:
I don't see how this effects any of that, there were already a bunch of methods with return types, now there are more. Feel free to PR whatever you want to change, not much use picking something apart that didn't break anything for you.